core/src/Streamly/Internal/Data/Binary/Decode.hs

```diff
@@ -266,6 +274,173 @@ word64leD = PRD.Parser step initial extract
 word64le :: Monad m => Parser Word8 m Word64
 word64le = PRD.toParserK word64leD
 
+{-# INLINE int8 #-}
+int8 :: Monad m => Parser Word8 m Int8
+int8 = PRD.toParserK $ PRD.Parser step initial extract
+
+    where
+
+    initial = return $ PRD.IPartial Nothing'
+
+    step _ a = return $ PRD.Done 0 (fromIntegral a :: Int8)
+
+    extract _ = return $ PRD.Error "int8: end of input"
```

Comment:
This should be simply `fromIntegral <$> word8`. Similarly, for other int types as well.